lib/mapper.js

```diff
@@ -47,6 +48,129 @@ exports.track = function(track){
 };
 
 /**
+ * Map Completed Order
+ *
+ * Docs: http://docs.klaviyo.com/article/60-ecommerce-integrating-a-custom-cart
+ *
+ * Must call Top level request for "Placed Order" and separate request for each item in products array
+ * @param {Track} track
+ * @return {Oject}
+ * @api private
+ */
+
+exports.completedOrder = function(track, settings){
+  var products = track.products();
+  var categories = formatCategories(products);
+  var productNames = formatNames(products);
+  var items = formatItems(products);
+
+  // TODO return two payloads
+  var payloads = {};
+
+  // Placed Order payload
+  payloads.order = {
+    token: settings.apiKey,
+    event: 'Placed Order',
+    time: time(track.timestamp())
+  };
+
+  payloads.order.customer_properties = {
+    $id: track.userId() || track.sessionId()
+  };
+
+  payloads.order.properties = {
+    $event_id: track.orderId(),
+    $value: track.revenue(),
+    Categories: categories,
+    'Item Names': productNames,
+    'Items': items
+  };
+
+  // Ordered Product payloads
+  payloads.products = products.map(function(item){
+    var product = new Track({ properties: item });
+    var payload = {
```

Comment:
A better name for this would be `productPayload`; you're using the term `payload` for a lot of things here, which makes reading through this pretty difficult